tests/test_csc.py

```diff
@@ -706,128 +652,113 @@ def __init__(self, *args, **kwargs):
+            harness.csc.log.warning(warn_message)
+            msg = await harness.remote.evt_logMessage.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(msg.message, warn_message)
+            self.assertEqual(msg.level, logging.WARNING)
+            self.assertEqual(msg.traceback, "")
+
+            with self.assertRaises(asyncio.TimeoutError):
+                await harness.remote.evt_logMessage.next(flush=False, timeout=NODATA_TIMEOUT)
 
-                msg = await harness.remote.evt_logMessage.next(flush=False, timeout=STD_TIMEOUT)
-                self.assertIn(harness.csc.exc_msg, msg.traceback)
-                self.assertIn("Traceback", msg.traceback)
-                self.assertIn("RuntimeError", msg.traceback)
-                self.assertEqual(msg.level, logging.ERROR)
+            await harness.remote.cmd_setLogLevel.set_start(level=logging.ERROR, timeout=STD_TIMEOUT)
 
-        asyncio.get_event_loop().run_until_complete(doit())
+            logLevel = await harness.remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(logLevel.level, logging.ERROR)
 
+            info_message = "test info message"
+            harness.csc.log.info(info_message)
+            with self.assertRaises(asyncio.TimeoutError):
+                await harness.remote.evt_logMessage.next(flush=False, timeout=NODATA_TIMEOUT)
 
-class BaseCscMainTestCase(unittest.TestCase):
+            warn_message = "test warn message"
+            harness.csc.log.warning(warn_message)
+            with self.assertRaises(asyncio.TimeoutError):
+                await harness.remote.evt_logMessage.next(flush=False, timeout=NODATA_TIMEOUT)
+
+            with salobj.assertRaisesAckError():
+                await harness.remote.cmd_wait.set_start(duration=5, timeout=STD_TIMEOUT)
+
+            msg = await harness.remote.evt_logMessage.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertIn(harness.csc.exc_msg, msg.traceback)
+            self.assertIn("Traceback", msg.traceback)
+            self.assertIn("RuntimeError", msg.traceback)
+            self.assertEqual(msg.level, logging.ERROR)
+
+
+class BaseCscMainTestCase(asynctest.TestCase):
     def setUp(self):
         salobj.set_random_lsst_dds_domain()
         self.original_argv = sys.argv[:]
 
     def tearDown(self):
         sys.argv = self.original_argv
 
-    def test_no_index(self):
-        async def doit(index):
-            sys.argv = [sys.argv[0]]
-            arg1 = "astring"
-            arg2 = 2.75
-            async with NoIndexCsc.main(index=index, arg1=arg1, arg2=arg2, run_loop=False) as csc:
-                self.assertEqual(csc.arg1, arg1)
-                self.assertEqual(csc.arg2, arg2)
-                await csc.do_exitControl(data=None)
-                await asyncio.wait_for(csc.done_task, timeout=5)
-
+    async def test_no_index(self):
         for index in (False, None):
             with self.subTest(index=index):
```

Comment:
Since there are only two cases here, I would suggest making them separate calls to `check_something_index`, to make it more obvious what is being tested/broken than the loop is.

Reply:
I prefer it the way it is. It is significantly more compact, which I think aids readability.

However, in looking at the code I realized I was not testing index=0 and the documentation was not sufficiently clear on what should happen. 0 did act like `None` and indeed it does.. I clarified the code and docs to make this more obvious and updated the test.

I no longer mention `False` as a valid value because it doesn't match the usage anywhere else and nobody is using it. However, it continues to work (both for backwards compatibility and because it's would require extra work to reject it).